src/test/java/edu/harvard/iq/dataverse/api/DatasetTypesIT.java

```diff
@@ -497,5 +635,82 @@ public void testLinkInstrumentToAstro() {
                 .body("data[1].fields.astroObject.displayOnCreate", equalTo(false));
 
     }
+    
+    @Test
+    public void testCreateDatasetWithCustomType() {
+        Response createUser = UtilIT.createRandomUser();
+        createUser.then().assertThat().statusCode(OK.getStatusCode());
+        String username = UtilIT.getUsernameFromResponse(createUser);
+        String apiToken = UtilIT.getApiTokenFromResponse(createUser);
+
+        Response createDataverse = UtilIT.createRandomDataverse(apiToken);
+        createDataverse.then().assertThat().statusCode(CREATED.getStatusCode());
+        String dataverseAlias = UtilIT.getAliasFromResponse(createDataverse);
+        Integer dataverseId = UtilIT.getDataverseIdFromResponse(createDataverse);
+       
+        UtilIT.setSuperuserStatus(username, true).then().assertThat().statusCode(OK.getStatusCode());
+   
+        JsonObjectBuilder job = Json.createObjectBuilder();
+        job.add("name", "testDatasetType");
+        job.add("linkedMetadataBlocks", Json.createArrayBuilder().add("geospatial"));
+        job.add("availableLicenses", Json.createArrayBuilder().add("CC0 1.0"));
+        
+        Response typeAdded = UtilIT.addDatasetType(job.build(), apiToken);
+        typeAdded.prettyPrint();
+
+        typeAdded.then().assertThat().statusCode(OK.getStatusCode());
+        Response getTypes = UtilIT.getDatasetTypes();   
+        getTypes = UtilIT.getDatasetTypes();
+        getTypes.prettyPrint();
+
+        String pathToJsonFile = "scripts/api/data/dataset-create-new-with-type.json";
+        
+        Response createDatasetResponse = UtilIT.createDatasetViaNativeApi(dataverseAlias, pathToJsonFile, apiToken);
+        
+        createDatasetResponse.prettyPrint();
+        
+
+        createDatasetResponse.then().assertThat().statusCode(CREATED.getStatusCode());
```

Comment:
I'm confused. Shouldn't this be a failure instead of CREATED? The only available license is CC0 1.0 but in the JSON "CC BY 4.0" is passed in. Shouldn't it complain that you have to pass CC0 1.0?